### DISCUSSION TITLE
Is useCookie reactive ?

### DISCUSSION THREAD

**Discussion by k0mar12** [nuxt/nuxt#17977] [Questions] [answered]
Hey there.

I need to pass `x-xsrf-token`, this token is locating in cookie `XSRF-TOKEN` which updating with every request to server.

So, i created my composable fetch:

```
export const useSpaFetch = () => {
    const token = useCookie('XSRF-TOKEN');

    const defaultOptions = () => {
        if (token.value === undefined) {
            return {}
        }

        return {
            headers: {
                'x-xsrf-token': token.value,
            }
        }
    };

    return {
        defaultOptions,
        useFetch: async (path, options) => useFetch(path, { ...defaultOptions(), ...options })
    }
};

```

The problem is, that my `token.value` is not reactive. Maybe its should be. I dont know exactly.
I mean, if i will make server request, `document.cookie` updating successfully but `token.value` not changing.

**Comment by danielroe** [comment, marked as answer]
`useCookie` is reactive, but your code fixes the cookie at the point in time you call `useFetch`. If you want the headers to be re-evaluated on every fetch you should likely do something like:

```diff
-        useFetch: async (path, options) => useFetch(path, { ...defaultOptions(), ...options })
+        useFetch: async (path, options) => useAsyncData(path, () => $fetch(path, { ...defaultOptions(), ...options }))
```

**Reply by k0mar12** [reply]
Yep, great thank You.
